### PR TITLE
Fix driver compilation for 5.11 kernel

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
> Made changes to fix compilation issues of driver with 5.11 kernel version.
> In 5.11 kernel drm_driver structure is changed, added changes in drm framework accordingly.
> Features missing : **p2p** support changes are missing.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 1]
```diff
@@ -537,6 +542,14 @@ static struct drm_driver mm_drm_driver = {
 	.date				= driver_date,
 };
 
+const struct drm_gem_object_funcs xocl_gem_object_funcs = {
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:545]: Please add 5.11 check here as well. I am not sure whether this gets compiled other Kernels or not.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 2]
```diff
@@ -537,6 +542,14 @@ static struct drm_driver mm_drm_driver = {
...
+        .free = xocl_free_object,
+        .get_sg_table = xocl_gem_prime_get_sg_table,
+        .vmap = xocl_gem_prime_vmap,
+        .vunmap = xocl_gem_prime_vunmap,
+        .vm_ops = &xocl_vm_ops,
+};
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:551]: Please maintain same order as < 5.11 sothat it is easy to compare.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 3]
```diff
@@ -537,6 +542,14 @@ static struct drm_driver mm_drm_driver = {
 	.date				= driver_date,
 };
 
+const struct drm_gem_object_funcs xocl_gem_object_funcs = {
+        .free = xocl_free_object,
+        .get_sg_table = xocl_gem_prime_get_sg_table,
+        .vmap = xocl_gem_prime_vmap,
+        .vunmap = xocl_gem_prime_vunmap,
+        .vm_ops = &xocl_vm_ops,
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:550]: gem_prime_export is missing from this ops. Please check.

**Review by chvamshi-xilinx — COMMENTED** [review]
Overall looks good. Couple of comments.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c (hunk 3 above)
Comment (rbramand-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c]: @chvamshi-xilinx made changes as suggested. Thanks

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good. Thanks

**Review by maxzhen — APPROVED** [review]

**Comment by keryell** [comment]
I need to bring some Champagne! :-) But I think it is unsuitable at work...

**Comment by HFTrader** [comment]
> I need to bring some Champagne! :-) But I think it is unsuitable at work...

Not in Russia or Germany...

**Pull request merged**